backend/capellacollab/projects/capellamodels/crud.py

```diff
@@ -4,52 +4,81 @@
 # Copyright DB Netz AG and the capella-collab-manager contributors
 # SPDX-License-Identifier: Apache-2.0
 
+# Standard library:
 import typing as t
 
 from fastapi import HTTPException
+
+# 3rd party:
 from sqlalchemy.orm import Session
 
+# 1st party:
+import capellacollab.projects.crud as projects_crud
+from capellacollab.projects.capellamodels.models import (
+    EmptyModel,
+    Model,
+    NewModel,
+    ToolDetails,
+)
 from capellacollab.projects.models import DatabaseProject
 from capellacollab.tools.models import Tool, Type, Version
 
-from .models import CapellaModelType, DB_CapellaModel, EditingMode, NewModel
-
 
-def get_all(db: Session, project_slug: str) -> t.List[DB_CapellaModel]:
+def get_all_models(db: Session, project_slug: str) -> t.List[Model]:
     project = (
         db.query(DatabaseProject)
         .filter(DatabaseProject.slug == project_slug)
         .first()
     )
-    return (
-        db.query(DB_CapellaModel)
-        .filter(DB_CapellaModel.project_name == project.name)
-        .all()
+    if not project:
+        raise HTTPException(404, detail="Project not found.")
+    return db.query(Model).filter(Model.project_id == project.id).all()
+
+
+def get_model_by_id(db: Session, id_: int) -> Model:
+    return db.query(Model).filter(Model.id == id_).first()
+
+
+def get_model_by_slug(db: Session, project_slug: str, slug: str) -> Model:
+    project = projects_crud.get_project_by_slug(db, project_slug)
+    model = (
+        db.query(Model)
+        .filter(
+            Model.project_id == project.id,
+            Model.slug == slug,
+        )
+        .first()
     )
+    if not model:
+        raise HTTPException(404, "Model not found.")
```

Comment:
Please use the custom error message style here and include a bit more context in the message, e.g. `The model with the name <slug> was not found.`